File: sample/app/src/main/java/com/example/activitystarter/StudentSerializableActivity.java
package com.example.activitystarter;

import android.os.Bundle;
import android.widget.TextView;

import activitystarter.Arg;
import activitystarter.MakeActivityStarter;

@MakeActivityStarter
public class StudentSerializableActivity extends BaseActivity {

    @Arg StudentSerializable student;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_data);
        TextView nameView = (TextView) findViewById(R.id.name_view);
        TextView idView = (TextView) findViewById(R.id.id_view);
        TextView gradeView = (TextView) findViewById(R.id.grade_view);
        TextView isPassingView = (TextView) findViewById(R.id.is_passing_view);

        nameView.setText("Name: "+student.name);
        idView.setText("Id: "+student.id);
        gradeView.setText("Grade: "+student.grade);
    }
}

File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/StudentSerializableActivity.kt
package com.marcinmoskala.kotlinapp

import activitystarter.Arg
import activitystarter.MakeActivityStarter
import android.os.Bundle
import android.widget.TextView
import com.marcinmoskala.activitystarter.argExtra

@MakeActivityStarter
class StudentSerializableActivity : BaseActivity() {

    @get:Arg val student: StudentSerializable  by argExtra()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_data)
        val nameView = findViewById(R.id.nameView) as TextView
        val idView = findViewById(R.id.idView) as TextView
        val gradeView = findViewById(R.id.gradeView) as TextView
        val isPassingView = findViewById(R.id.isPassingView) as TextView

        nameView.text = "Name: ${student.name}"
        idView.text = "Id: ${student.id}"
        gradeView.text = "Grade: ${student.grade}"
        isPassingView.text = "Passing: ${student.passing}"
    }
}
